File: 3066/sol.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;

class Solution {
public:
  int minOperations(vector<int>& nums, int k) {
    multiset<ll> s;
    for (int x: nums) s.insert(x);
    int res = 0;
    while((*s.begin())<k) {
      auto it = s.begin();
      ll x = *it;
      it++;
      ll y = *it;
      s.erase(s.begin());
      s.erase(s.begin());
      s.insert(2*1ll*x+y);
      res++;
    }
    return res;
  }
};

File: 3191/sol.py
class Solution(object):
  def minOperations(self, nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    res = 0
    for i in range(len(nums)-2):
      if nums[i] == 0:
        res += 1
        for j in range(3):
          nums[i+j] = 1 - nums[i+j]
    return -1 if 0 in nums else res
        